src/node_process_methods.cc

```diff
@@ -171,6 +171,12 @@ static void Kill(const FunctionCallbackInfo<Value>& args) {
   if (!args[0]->Int32Value(context).To(&pid)) return;
   int sig;
   if (!args[1]->Int32Value(context).To(&sig)) return;
+    // TODO(joyeecheung): white list the signals?
+
+#if HAVE_INSPECTOR
+  profiler::EndStartedProfilers(env);
```

Comment:
@joyeecheung Should this be guarded to only be run if `sig > 0` and `pid` is one of `0, -1, getpid(), -getpid()` or something similar?

Reply:
I think we should only write the profiles when we are sure the kill is initiated to terminate the process, are those conditions enough for us to be sure about that?

Reply:
@joyeecheung I mean, some signals that would usually terminate the process can be caught (like `SIGINT`), so we cannot be sure of whether this is going to terminate the process or not no matter what…

Reply:
Maybe we can only take care of signals that don't have a handler attached through Node.js's APIs, though even then it would get complicated because of workers. Or we could just not guard this at all, or let the user decide somehow, but then no one has reported any issues with this so I don't know what the real-world use cases would be.